webapp/tests/karma/ts/modules/messages/messages.component.spec.ts

```diff
@@ -172,13 +191,117 @@ describe('Messages Component', () => {
       expect(component.loading).to.be.false;
       expect(component.conversations).to.eql( newConversations);
     });
+
+    it('ngOnDestroy() should unsubscribe from observables', () => {
+      const spySubscriptionsUnsubscribe = sinon.spy(component.subscriptions, 'unsubscribe');
+
+      component.ngOnDestroy();
+
+      expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
+    });
   });
 
-  it('ngOnDestroy() should unsubscribe from observables', () => {
-    const spySubscriptionsUnsubscribe = sinon.spy(component.subscriptions, 'unsubscribe');
+  describe('Messages breadcrumbs', () => {
+    const bettyOfflineUserContactDoc = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'CHW Bettys Area',
+        parent: userContactGrandparent,
+      },
+    };
+    const conversations =  [
+      { key: 'a',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area', null]
+      },
+      { key: 'b',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village']
+      },
+      { key: 'c',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'Ramdom Place', null, null]
+      },
+      { key: 'd',
+        message: { inAllMessages: true },
+        lineage : []
+      },
+      { key: 'e',
+        message: { inAllMessages: true },
+      },
+    ];
+
+    it('it should retrieve the hierarchy level of the connected user', async () => {
+      expect(await component.currentLevel).to.equal('parent');
+    });
+
+    it('should not change the conversations lineage if the connected user is online only', fakeAsync( () => {
```

Comment:
```suggestion
    it('should not change the conversations lineage if the connected user is online only', fakeAsync(() => {
```
Same with the other occurrences of `fakeAsync`